bin/critterget.py

```diff
@@ -563,19 +563,23 @@ def getCredentials(sessionKey):
 
     if (debug) : print u'{} MessageType="ApteligentDebug"  Into getCredentials'.format(DATETIME_OF_RUN)
 
-    try:
-        # list all credentials
-        entities = entity.getEntities(['admin', 'passwords'], namespace=myapp,
-                                    owner='nobody', sessionKey=sessionKey)
-    except Exception, e:
-        print u'{} MessageType="ApteligentDebug" Could not get {} credentials from splunk. Error: {}'.format(DATETIME_OF_RUN, myapp, str(e))
-
-    # return first set of credentials
-    if (debug) : print "Entities is ", entities
-    for i, c in entities.items():
-        return c['clear_password']
-
-    print u'{} MessageType="ApteligentDebug" No credentials have been found for app {} . Maybe a setup issue?'.format(DATETIME_OF_RUN, myapp)
+    auth = None
+
+    while auth is None:
+        try:
+            # list all credentials
+            entities = entity.getEntities(['admin', 'passwords'], namespace=myapp,
+                                        owner='nobody', sessionKey=sessionKey)
+        except Exception, e:
+            print u'{} MessageType="ApteligentDebug" Could not get {} credentials from splunk. Error: {}'.format(DATETIME_OF_RUN, myapp, str(e))
+
+        # return first set of credentials
+        if (debug) : print "Entities is ", entities
+        for i, c in entities.items():
```

Comment:
You're still not doing the sanity check here.  Since your default value for `entities` is `None`, think through what will happen here if you arrive here and `entities` is still `None`.